### PR TITLE
AQL: Add REPLACE_NTH() array function

### PR DESCRIPTION
**Pull request by dothebart**
https://github.com/arangodb/arangodb/pull/10955

https://jenkins01.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/565/

### CODE REVIEW

File: 3.7/aql/functions-array.md [hunk 1]
```diff
@@ -536,6 +536,43 @@ RETURN REMOVE_NTH( [ "a", "b", "c", "d", "e" ], -2 )
 {% endaqlexample %}
 {% include aqlexample.html id=examplevar type=type query=query bind=bind result=result %}
 
+
+## REPLACE_NTH()
+
+`REPLACE_NTH(anyArray, position, replaceValue, defaultPaddValue) → newArray`
```
Comment (jsteemann) [inline 3.7/aql/functions-array.md]:
```suggestion
`REPLACE_NTH(anyArray, position, replaceValue, defaultPaddingValue) → newArray`
```

File: 3.7/aql/functions-array.md [hunk 2]
```diff
@@ -536,6 +536,43 @@ RETURN REMOVE_NTH( [ "a", "b", "c", "d", "e" ], -2 )
...
+
+Replace the element at *position* from the *anyArray* with *replaceValue*. If *defaultPaddValue* is specified, and *position* points after an already existing entry in *anyArray*, *defaultPaddValue* is inserted as many times as needed to place *replaceValue* at the end of *annyArray*
```
Comment (jsteemann) [inline 3.7/aql/functions-array.md]:
```suggestion
Replace the element at *position* from the *anyArray* with *replaceValue*. If *defaultPaddingValue* is specified, and *position* points behind an already existing entry in *anyArray*, *defaultPaddingValue* is inserted as many times as needed to place *replaceValue* at the end of *anyArray*
```

File: 3.7/aql/functions-array.md [hunk 3]
```diff
@@ -536,6 +536,43 @@ RETURN REMOVE_NTH( [ "a", "b", "c", "d", "e" ], -2 )
...
+
+- **anyArray** (array): array with elements of arbitrary type
+- **position** (number): the position of the element to replace. Positions start
+  at 0. Negative positions are unsupported.
+  If *position* is out of bounds, and *defaultPaddValue* is not specified,
```
Comment (jsteemann) [inline 3.7/aql/functions-array.md]:
```suggestion
  If *position* is out of bounds, and *defaultPaddingValue* is not specified, 
```

**Comment by Simran-B** [comment]
Hm, not sure why it fails even though the error is expected (xpError)...
```
done with  aqlArrayReplaceNth_3
ERROR JavaScript exception in file '…js/client/modules/@arangodb/arangosh.js' at 99,7:
ArangoError 1541: AQL: invalid number of arguments for function 'REMOVE_NTH()',
expected number of arguments: minimum: 2, maximum: 2 (while parsing)
```

**Comment by Simran-B** [comment]
https://deploy-preview-319--zealous-morse-14392b.netlify.com/docs/3.7/aql/functions-array.html#replace_nth

Not great to use AQL examples except for the one with an expected error, for which arangosh needs to be used, and the spacing between command and output is the same as between the other examples, but for now this is fine.

**Pull request merged**